src/cli/argument_parser_facade.py

```diff
@@ -0,0 +1,129 @@
+import argparse
+from abc import ABC, abstractmethod
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Any, Optional, Sequence
+
+from src.cli.cli_app import CLIApp
+from src.cli.command import Command
+
+
+class ArgumentDefaultValueProvider(ABC):
```

Comment:
Why do we need this base class? It seems only one class implements it

Reply:
I used Delegate Pattern here. I wanted to avoid making `ArgumentParserFacade` dependent on the `Project` class.

Reply:
I understand why we have `ArgumentDefaultValueFromConfigProvider`, we may need some mapping between project and facade, which is fine. I don't understand the purpose of an abstract base class, imo it's unnecessary abstraction for now, we can isolate it later when we will have more ValueProviders

Reply:
This abstract class was meant to be an interface. This class is implemented in the test file (`argument_parser_facade_test`).

Reply:
Python is duck typed, interfaces exist only for documentation reasons. Having an interface makes sense when you have multiple implementations of it. Here we introduce confusing and premature abstraction which makes code harder to comprehend. We have nothing to generalize. Please remove the interface and use class directly. For test you can use mock.

Reply:
The point of this class is not an abstraction nor generalization, but making the code loosely coupled. I wanted to make the `cli` package completely independent of the `project`.

Fine, I'll remove it.